Source/Common/GLTFSceneSource.swift

```diff
@@ -11,49 +11,14 @@ import SceneKit
 @objcMembers
 @available(iOS 10.0, *)
 public class GLTFSceneSource : SCNSceneSource {
-    private var loader: GLTFUnarchiver! = nil
+    private let loader: GLTFUnarchiver
     
-    public override init() {
-        super.init()
-    }
-    
-    public convenience init(path: String, options: [SCNSceneSource.LoadingOption : Any]? = nil, extensions: [String:Codable.Type]? = nil) throws {
-        self.init()
-        
-        let loader = try GLTFUnarchiver(path: path, extensions: extensions)
-        self.loader = loader
-    }
-    
-    public override convenience init(url: URL, options: [SCNSceneSource.LoadingOption : Any]? = nil) {
-        self.init()
-        do {
-            let loader = try GLTFUnarchiver(url: url, extensions: nil)
-            self.loader = loader
-        } catch {
-            print("\(error.localizedDescription)")
-        }
-    }
-    
-    public convenience init(url: URL, options: [SCNSceneSource.LoadingOption : Any]?, extensions: [String:Codable.Type]?) throws {
-        self.init()
-        self.loader = try GLTFUnarchiver(url: url, extensions: extensions)
-    }
-    
-    public override convenience init(data: Data, options: [SCNSceneSource.LoadingOption : Any]? = nil) {
-        self.init()
-        do {
-            self.loader = try GLTFUnarchiver(data: data)
-        } catch {
-            print("\(error.localizedDescription)")
+    public init?(url: URL, options: [SCNSceneSource.LoadingOption : Any]?, extensions: [String:Codable.Type]?) {
```

Comment:
The `SCNSceneSource` initializers are failable, not throwable so we can't have a throwable initializer delegate to a failable initializer. The original version had the `public override init()` and implicitly unwrapped `loader` to bypass this. By using a failable initializer, we can delegate directly to the `SCNSceneSource` designated init and ensure that the `loader` object is set at the end of initialization of the `GLTFSceneSource` object.

Reply:
LGTM